### PR TITLE
Update sir

### PR DESCRIPTION
**Pull request by marioiordanov**
## Development related changes

- Updated starknet_in_rust version to [3febaf6](https://github.com/lambdaclass/starknet_in_rust/commit/3febaf633bfa92caa26d2e1159a503d332b95088).

## Necessary changes

- Modified state_diff implementation, due to this https://github.com/lambdaclass/starknet_in_rust/blob/1eb0411cc875c17c4e176d8411dbf27bb81e3d24/src/transaction/declare_v2.rs#L378
- Changed unit test
- Added unit test

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [ ] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet/src/starknet/add_deploy_account_transaction.rs [hunk 1]
```diff
@@ -121,10 +121,47 @@ mod tests {
...
+            &vec![],
+            Fee(fee_raw),
+            &vec![],
+            Felt::from(0),
+            account_class_hash,
+            Felt::from(13),
+            Felt::from(1),
+        );
+
+        let sir_transction = transaction
```
Comment (FabijanC) [inline crates/starknet/src/starknet/add_deploy_account_transaction.rs:143]: typo: transction

File: crates/starknet/src/starknet/add_deploy_account_transaction.rs [hunk 2]
```diff
@@ -121,10 +121,47 @@ mod tests {
         let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
         let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
 
+        assert_eq!(txn.finality_status, None);
+        assert!(txn.execution_result.revert_reason().unwrap().contains("Fee transfer failure"));
+    }
+
+    #[test]
+    fn deploy_account_transaction_should_fail_due_to_not_enough_fee() {
```
Comment (FabijanC) [inline crates/starknet/src/starknet/add_deploy_account_transaction.rs:129]: What is the difference between this and the previous test (`deploy_account_transaction_should_fail_due_to_low_balance`)?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: crates/starknet/src/starknet/add_deploy_account_transaction.rs (hunk 2 above)
Comment (marioiordanov) [reply crates/starknet/src/starknet/add_deploy_account_transaction.rs:129]: In this test the account that is making the transaction have some balance in fee_token_contract, but due to the low max fee that it provided, the transaction fails and the max fee is being charged.
In the other one the account doesn't have any balance, so when it tries to charge with the amount = max_fee, it returns an error, something like an exception from starknet_in_rust

File: crates/starknet/src/starknet/add_deploy_account_transaction.rs (hunk 1 above)
Comment (marioiordanov) [reply crates/starknet/src/starknet/add_deploy_account_transaction.rs]: https://github.com/0xSpaceShard/starknet-devnet-rs/commit/2eb632c3a3230bf568b98ddc0b3f7d974e5d404f

**Review by FabijanC — APPROVED** [review]

**Pull request merged**